src/redisearch_rs/inverted_index_bencher/src/bencher.rs

```diff
@@ -96,51 +45,313 @@ impl NumericBencher {
+        BenchEncodingInputs {
+            group: "Float - Big Neg",
+            values: vec![-1.1, -0.3, -12.999999999999998],
+            value_size_fn: |_| 8,
+        },
+    ];
+
+    let deltas = vec![
+        // 0 bytes
+        0,
+        // 4 bytes
+        100_000_000,
+        // 7 bytes
+        1_000_000_000_000_000,
+    ];
+
+    encoding_values
+        .into_iter()
+        .cartesian_product(deltas)
+        .map(
+            |(
+                BenchEncodingInputs {
+                    group,
+                    values,
+                    value_size_fn,
+                },
+                delta,
+            )| {
+                let inputs = values
+                    .into_iter()
+                    // We need to find the actual resulting output for the decoding benchmarks
+                    .map(|value| {
+                        let record = inverted_index::RSIndexResult::numeric(0, value);
+                        let mut buffer = Cursor::new(Vec::new());
+                        let _grew_size =
+                            Numeric::encode(&mut buffer, Delta::new(delta), &record).unwrap();
+                        let buffer = buffer.into_inner();
+
+                        (value, delta, buffer)
+                    })
+                    // Find the input and delta sizes in bytes to group the benchmarks
+                    .map(|(value, delta, buffer)| {
+                        let value_size = value_size_fn(value);
+                        let delta_size = (((delta + 1) as f64).log2() / 8.0).ceil() as usize;
+
+                        assert_eq!(
+                            buffer.len(),
+                            1 + delta_size + value_size,
+                            "the wrong size calculation is used for group: {group}"
+                        );
+                        (value, delta, buffer, value_size, delta_size)
+                    })
+                    .fold(
+                        HashMap::new(),
+                        |mut map, (value, delta, buffer, value_size, delta_size)| {
+                            map.entry((value_size, delta_size))
+                                .or_insert(vec![])
+                                .push((value, delta, buffer));
 
-                    black_box(grew_size);
+                            map
+                        },
+                    )
+                    .into_iter()
+                    .map(|((value_size, delta_size), values)| BenchInput {
+                        values,
+                        delta_size,
+                        value_size,
+                    })
+                    .collect();
+
+                BenchGroup {
+                    group: group.to_string(),
+                    inputs,
                 }
             },
-            BatchSize::SmallInput,
-        );
-    });
+        )
+        .collect()
+}
+
+fn numeric_c_encode<M: Measurement>(group: &mut BenchmarkGroup<'_, M>, input: &BenchInput) {
```

Comment:
In my version I made those functions methods of the Bencher object. I think it's cleaner so each bencher as its stuffs properly namespaced and grouped together if they are all defined in this file.

Or we could just move them to their own file as you suggested above.

Reply:
I think it will be easier to navigate if each is in its own file. I'll move mine in this PR